src/lib/dot-matrix/dotMatrix.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState, useEffect } from "react";
+import { v4 } from 'uuid';
+import classes from './styles.module.scss';
+import { DotMatrixPropType, DataPointType } from "./types";
+import { COLOR_PALATTE, Elements } from "./constants";
+
+const DotMatrix = (props: DotMatrixPropType): JSX.Element => {
+  const {
+    title,
+    dataPoints,
+    rows = 5,
+    columns = 12,
+    styles = {}
+  } = props;
+  const [data, setData] = useState<DataPointType[]>([]);
+  const [totalVal, setTotal] = useState<number>(0);
+
+  const [partialVal, setPartialVal] = useState<number[]>([]);
+  const getStyles = (element: Elements): object => {
+    const getElementStyle = styles[element];
+    if (getElementStyle) {
+      return getElementStyle();
+    }
+    return {};
+  };
+  useEffect(() => {
+    if (dataPoints) {
+      const values = [...(data || [])];
+      let total = 0
+      let colorIndex = 0;
+      dataPoints.map((point: DataPointType, indexVal) => {
+        total += point.count;
+        if (point.colorPreference) {
+          values.push({ ...point});
+        } else {
+          let randomColor = ''
+          do {
+            randomColor = COLOR_PALATTE[colorIndex];
+            colorIndex++;
+          } while (randomColorPresent(randomColor, values))
+          values.push({ ...point, colorPreference: randomColor})
+        }
+        if (indexVal === dataPoints.length - 1) {
+          setTotal(total);
+          setData(values);
+        }
+      })
+    }
+  }, [dataPoints]);
+
+  useEffect(() => {
+    const partial: Array<number> = [];
+    if (totalVal) {
+      data?.map((each: DataPointType, i: number) => {
+        partial.push(getPartialDots(each, totalVal));
+        if (i === data?.length - 1) setPartialVal(partial);
+      })
+    }
+  }, [totalVal]);
+  const randomColorPresent = (color: string, dataValues: DataPointType[] = []): boolean => {
```

Comment:
For a function returning a boolean it is better to have `is`, `has`, etc as prefix?